iosApp/iosApp/SceneDelegate.swift

```diff
@@ -45,7 +45,24 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
         // Use this method to save data, release shared resources, and store enough scene-specific state information
         // to restore the scene back to its current state.
     }
-
-
+    
+    func scene(_ scene: UIScene, openURLContexts URLContexts: Set<UIOpenURLContext>) {
```

Comment:
Thinking it might be possible refactor some of the nested `if` conditionals here. I think the following is logically consistent but coalesces some of the `if`'s to reduce the nesting.

```
    func scene(_ scene: UIScene, openURLContexts URLContexts: Set<UIOpenURLContext>) {
        guard let urlContext = URLContexts.first else {
            return
        }

        if let components = URLComponents(url: urlContext.url, resolvingAgainstBaseURL: false),
           let queryItems = components.queryItems {
            for item in queryItems {
                if item.name == "code", let code = item.value {
                    if let windowScene = scene as? UIWindowScene,
                       let rootViewController = windowScene.windows.first?.rootViewController as? TestbedViewController {
                        rootViewController.setAuthCode(code)
                    }
                }
                break
            }
        }
    }
```